### PR TITLE
Resolves issue #951 by adding support for the Red Hat OpenShift on AWS (rosa) cli tool

### PR DESCRIPTION
**Pull request by mkimberley**
- Adds a new tool "rosa" which is the cli for Red OpenShift on AWS

## Description
- Modified tools.go, README.md and get_test.go to add the new tool

## 
Motivation and Context
- Resolves issue #951 
- [x] I have raised an issue to propose this change, which has been given a label of `design/approved` by a maintainer ([required](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?
- make e2e
- using test-tool.sh

If updating or adding a new CLI to `arkade get`, run:

```
go build && ./hack/test-tool.sh TOOL_NAME
```

```
./test-tool.sh rosa
+ ./arkade get rosa --arch arm64 --os darwin --quiet
+ file /Users/matt/.arkade/bin/rosa
/Users/matt/.arkade/bin/rosa: Mach-O 64-bit executable x86_64
+ rm /Users/matt/.arkade/bin/rosa
+ echo

+ ./arkade get rosa --arch x86_64 --os darwin --quiet
+ file /Users/matt/.arkade/bin/rosa
/Users/matt/.arkade/bin/rosa: Mach-O 64-bit executable x86_64
+ rm /Users/matt/.arkade/bin/rosa
+ echo

+ ./arkade get rosa --arch x86_64 --os linux --quiet
+ file /Users/matt/.arkade/bin/rosa
/Users/matt/.arkade/bin/rosa: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=zc0zaJD6A5pZUpnZvzxM/EuR4UKJoDl1spt81-lsG/kpGc4hRgdWEXBbKPnqUc/aLqW7g4ULuz5Gqw-PDm0, not stripped
+ rm /Users/matt/.arkade/bin/rosa
+ echo

+ ./arkade get rosa --arch aarch64 --os linux --quiet
+ file /Users/matt/.arkade/bin/rosa
/Users/matt/.arkade/bin/rosa: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=zc0zaJD6A5pZUpnZvzxM/EuR4UKJoDl1spt81-lsG/kpGc4hRgdWEXBbKPnqUc/aLqW7g4ULuz5Gqw-PDm0, not stripped
+ rm /Users/matt/.arkade/bin/rosa
+ echo

+ ./arkade get rosa --arch x86_64 --os mingw --quiet
+ file /Users/matt/.arkade/bin/rosa.exe
/Users/matt/.arkade/bin/rosa.exe: PE32+ executable (console) x86-64 (stripped to external PDB), for MS Windows
+ rm /Users/matt/.arkade/bin/rosa.exe
+ echo
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation

- [x] I have updated the list of tools in README.md if (required) with `./arkade get --format markdown`
- [ ] I have updated the list of apps in README.md if (required) with `./arkade install --help`

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -836,6 +836,7 @@ There are 56 apps that you can install on your cluster.
 | [promtool](https://github.com/prometheus/prometheus)                         | Prometheus rule tester and debugging utility                                                                                              |
 | [rekor-cli](https://github.com/sigstore/rekor)                               | Secure Supply Chain - Transparency Log                                                                                                    |
 | [replicated](https://github.com/replicatedhq/replicated)                     | CLI for interacting with the Replicated Vendor API                                                                                        |
+| [rosa](https://github.com/openshift/rosa)                                    | Red Hat OpenShift on AWS (ROSA) command line tool                                                                                         |
```
Comment (Jasstkn) [inline README.md:839]: Hi! Could you regenerate the table? There must be one more line with the changed numbers of tools.

File: pkg/get/get_test.go [hunk 2]
```diff
@@ -6350,6 +6350,46 @@ func Test_DownloadOpenshiftCLI(t *testing.T) {
...
+	tools := MakeTools()
+	name := "rosa"
+
+	tool := getTool(name, tools)
+
+	const toolVersion = "1.2.23"
+
+	tests := []test{
+		{
+			os:      "linux",
```
Comment (Jasstkn) [inline pkg/get/get_test.go:6363]: I think you are missing a few tests here for different archs. e.g. https://github.com/alexellis/arkade/blob/master/pkg/get/get_test.go#L6467

File: pkg/get/tools.go [hunk 3]
```diff
@@ -3669,6 +3669,41 @@ https://github.com/{{.Owner}}/{{.Repo}}/releases/download/{{.Version}}/{{.Repo}}
...
+								{{$ext = "zip"}}
+							{{- end -}}
+		
+							{{- if eq .Arch "aarch64" -}}
+								{{$arch = "-arm64"}}
+							{{- else if eq .Arch "arm64" -}}
+								{{ $arch = "-arm64" }}
+							{{- else if eq .Arch "x86_64" -}}
+								{{ $arch = "" }}
+							{{- end -}}
```
Comment (Jasstkn) [inline pkg/get/tools.go:3697]: arch is not being used in the template, why do we need it?

**Comment by alexellis** [comment]
@mkimberley when do you expect to be able to check out Maria's comments? Let me know and I'll try to come back then to check again.

**Comment by mkimberley** [comment]
> @mkimberley when do you expect to be able to check out Maria's comments? Let me know and I'll try to come back then to check again.

Hi Alex, appreciate the feedback from Maria. I should be able to get this done in the next two weeks, just working on a project for work atm

**Pull request closed without merging**